### PR TITLE
Functional tests in separate processes - Subtask #851 - Close #863 

### LINKED ISSUE #863: Functional tests executed in separate processes
As we do in unit test, it is recommended to isolate each functional test into separate process to avoid conflicts.

Parent #851

### PR DESCRIPTION
**Pull request by diego-G**
Close task list #851 
Close #863

### CODE REVIEW

File: test/functional/http/get/index.js [hunk 1]
```diff
@@ -1,10 +1,50 @@
...
+	'./multisignatures',
+	'./multisignatures.post',
+	'./node',
+	'./peers',
+	'./transactions',
+];
+
+var parallelTestsRunning = {};
+
+paths.forEach(function (test) {
```
Comment (MaciejBaj) [inline test/functional/http/get/index.js]: The code responsible for running tests parallel should be reused as it is done now in 3 places

File: test/functional/http/post/index.js [hunk 2]
```diff
@@ -1,4 +1,44 @@
-require('./0.transfer');
-require('./1.second.secret');
-require('./2.delegate');
-require('./3.votes');
+'use strict';
+
+var cluster = require('cluster');
```
Comment (MaciejBaj) [inline test/functional/http/post/index.js]: unused dependency

File: test/functional/http/get/index.js [hunk 3]
```diff
@@ -1,10 +1,50 @@
...
-require('./delegates');
-require('./loader');
-require('./multisignatures');
-require('./multisignatures.post');
-require('./node');
-require('./peers');
-require('./transactions');
+'use strict';
+
+var cluster = require('cluster');
```
Comment (MaciejBaj) [inline test/functional/http/get/index.js]: unused dependency

File: test/functional/http/post/0.transfer.js [hunk 4]
```diff
@@ -129,7 +129,7 @@ describe('POST /api/transactions (type 0) transfer funds', function () {
 		it('sending already confirmed transaction should fail', function () {
 			return sendTransactionPromise(goodTransaction).then(function (res) {
 				node.expect(res).to.have.property('success').to.be.not.ok;
-				node.expect(res).to.have.property('message').to.equal('Transaction is already confirmed: ' + goodTransaction.id);
+				node.expect(res).to.have.property('message').to.have.string(goodTransaction.id);
```
Comment (MaciejBaj) [inline test/functional/http/post/0.transfer.js]: `.to.match(/^Transaction is already (confirmed|processed): [A-F0-9]{20}/);`

File: test/functional/http/post/index.js [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
+var parallelTests = require('../../../common/parallelTests').parallelTests;
+
+var pathFiles = [
+	'./0.transfer',
+	'./1.second.secret',
+	'./2.delegate',
+	'./3.votes',
+];
+
+parallelTests(pathFiles, 'test/functional/http/post/');
```
Comment (MaciejBaj) [inline test/functional/http/post/index.js]: missing empty line

File: test/unit/index.js [hunk 6]
```diff
@@ -52,35 +51,4 @@ var paths = [
...
-		console.log('Test failed:', test);
-		process.exit(code);
-	});
-
-	child.on('error', function (err) {
-		console.error(err);
-		process.exit();
-	});
-});
+parallelTests(pathFiles, 'test/unit/');
```
Comment (MaciejBaj) [inline test/unit/index.js]: missing empty line

File: test/functional/ws/index.js [hunk 7]
```diff
@@ -1,6 +1,14 @@
...
+var pathFiles = [
+	'./transport',
+	'./transport.blocks',
+	'./transport.client',
+	'./transport.handshake',
+	'./transport.transactions',
+	'./transport.transactions.stress',
+];
+
+parallelTests(pathFiles, 'test/functional/ws/');
```
Comment (MaciejBaj) [inline test/functional/ws/index.js]: missing empty line

File: test/common/parallelTests.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+		child.on('close', function (code) {
+			if (code === 0) {
+				console.log('Test finished successfully:', test);
+				delete parallelTestsRunning[child.pid];
+				if (Object.keys(parallelTestsRunning).length === 0) {
+					return console.log('All tests finished successfully.');
+				}
+				return console.log('Still running: ', parallelTestsRunning);
+			}
+			console.log('Test failed:', test);
```
Comment (Isabello) [inline test/common/parallelTests.js]: console log?

File: test/functional/http/post/index.js [hunk 9]
```diff
@@ -1,4 +1,4 @@
 require('./0.transfer');
 require('./1.second.secret');
-require('./2.delegate');
 require('./3.votes');
+require('./2.delegate');
```
Comment (Isabello) [inline test/functional/http/post/index.js]: why is this out of order?

**Pull request merged**